remote/remote_state_s3.go

```diff
@@ -529,7 +529,7 @@ func checkIfS3BucketNeedsUpdate(s3Client *s3.S3, config *ExtendedRemoteStateConf
 	}
 
 	if !config.SkipBucketSSEncryption {
-		enabled, err := checkIfSSEForS3Enabled(s3Client, &config.remoteStateConfigS3, terragruntOptions)
+		enabled, err := checkIfSSEForS3Enabled(s3Client, config, terragruntOptions)
```

Comment:
config.BucketSSEAlgorithm is not included in remoteStateConfig and we want to verify that the bucket is being set to match a user's config.BucketSSEAlgorithm